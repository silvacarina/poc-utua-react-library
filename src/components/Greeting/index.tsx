type GreetingProps = {
  greeting?: string
}

export default function Greeting({ greeting = "World" }: GreetingProps) {
  return (
    <div>Hello, {greeting}!</div>
  )
}