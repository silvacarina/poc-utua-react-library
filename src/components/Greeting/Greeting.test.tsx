import renderer from 'react-test-renderer'
import { describe, expect, it } from 'vitest'
import Greeting from './index'

describe('Greeting component', () => {
  it('Greeting component renders correctly', () => {
    const component = renderer.create(
      <Greeting />
    )

    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it(' prop working', () => {
    const component = renderer.create(
      <Greeting greeting="Universe" />
    )

    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})