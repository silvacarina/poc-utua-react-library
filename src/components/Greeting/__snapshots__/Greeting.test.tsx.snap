// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Greeting component >  prop working 1`] = `
<div>
  Hello, 
  Universe
  !
</div>
`;

exports[`Greeting component > Greeting component renders correctly 1`] = `
<div>
  Hello, 
  World
  !
</div>
`;
